test/integration/envtest/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the Gardener Scheduler
+
+	Prerequisites
+		- The Gardener-Scheduler is running
+
+	BeforeSuite
+		- Parse valid Shoot from example folder and flags. Remove the Spec.SeedName.
+		- If running in TestMachinery: Scale down the GardenerController Manager
+
+	AfterSuite
+		- Delete Shoot
+        - If running in TestMachinery: Scale up the GardenerController Manager
+
+	Test: SameRegion Scheduling Strategy Test
+		1) Create Shoot in region where no Seed exists. (e.g Shoot in eu-west-1 and only Seed exists in us-east-1)
+		   Expected Output
+			 - should fail because no Seed in same region exists1)
+	Test: Minimal Distance Scheduling Strategy Test
+		1) Create Shoot in region where no Seed exists. (e.g Shoot in eu-west-1 and only Seed exists in us-east-1)
+		   Expected Output
+			 - should successfully schedule to Seed in region with minimal distance
+	Test: Api Server ShootBindingStrategy test
+		1) Request APiServer to schedule shoot to non-existing seed
+		   Expected Output
+			 - Error from ApiServer
+		2) Request APiServer to schedule shoot that is already scheduled to another seed
+		   Expected Output
+			 - Error from ApiServer
+ **/
+
+package scheduler
+
+import (
+	"context"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/scheduler/apis/config"
+	shootcontroller "github.com/gardener/gardener/pkg/scheduler/controller/shoot"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
```

Comment:
```suggestion

	corev1 "k8s.io/api/core/v1"
	"k8s.io/apimachinery/pkg/api/resource"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/utils/pointer"
	"sigs.k8s.io/controller-runtime/pkg/client"
	"sigs.k8s.io/controller-runtime/pkg/manager"
```

Nit: Typically, we group the imports in 3 groups: (1) core golang packages, (2) local packages, (3) external packages